src/contentful/contentful.service.ts

```diff
@@ -0,0 +1,34 @@
+import { createClient, DeletedEntry, Entry } from 'contentful'
+import { Injectable } from '@nestjs/common'
+
+const { CONTENTFUL_SPACE_ID, CONTENTFUL_ACCESS_TOKEN } = process.env
+
+const contentfulClient = createClient({
+  space: CONTENTFUL_SPACE_ID,
+  accessToken: CONTENTFUL_ACCESS_TOKEN,
+})
+
+// TODO: Move this to a key/value store to persist across application instances
+let nextSyncToken = ''
+
+@Injectable()
+export class ContentfulService {
+  async getSync(
+    initial = false,
+  ): Promise<[entries: Entry[], deletedEntries: DeletedEntry[]]> {
+    const {
+      entries,
+      deletedEntries,
+      nextSyncToken: newToken,
+    } = await contentfulClient.sync({
+      ...(initial || !nextSyncToken ? { initial: true } : { nextSyncToken }),
+    })
+    nextSyncToken = newToken
+    return [entries, deletedEntries]
```

Comment:
I would prefer to return here an object
```
{ entries, deletedEntries }
```
so it is clear what is return and we don't need to know that the first object is the entries and the second one is the deleted entries.